### PR TITLE
Gavin/koordinates case study

### PR DESCRIPTION
**Pull request by gavination**
Adds the Koordinates case study draft as a blog post. Machine embeds might need to be replaced with screenshots as they're pretty dense.

### CODE REVIEW

File: blog/2023-11-15-koordinates-and-stately-case-study/index.mdx [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+In the case of Kart, many of these operations can be long-running and have **several** edge cases, making Stately’s tooling to visualize, understand, and execute logic from stateful diagrams a massive benefit.
+
+## Breaking complex UI’s into manageable components **
+
+In the case of both their products, Koordinates was able to use Stately’s tooling to break their “big” problems down into several, distinct smaller ones. Taylor Lodge, an engineer on the core team, explained that with Stately’s visual editor and simulation tooling, they were able to break their application logic into several small, testable state machines. For example, the UI for Koordinates Cloud has several components, but the logic for loading elements in their list view is represented rather succinctly in this state machine: 
+
+
+<EmbedMachine
+  name="Koordinates List View Machine"
+  embedURL="https://stately.ai/registry/editor/1b48bbf1-6f7a-4847-b1e0-47ee5c283371?mode=design&machineId=68b835cc-49fc-4b9c-972e-a63c01465c1f"/>
```
Comment (mellson) [inline blog/2023-11-15-koordinates-and-stately-case-study/index.mdx:44]:
```suggestion
  embedURL="https://stately.ai/registry/editor/embed/1b48bbf1-6f7a-4847-b1e0-47ee5c283371?machineId=68b835cc-49fc-4b9c-972e-a63c01465c1f&mode=design"/>
```

File: blog/2023-11-15-koordinates-and-stately-case-study/index.mdx [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+While there are other state management platforms for developers, the team at Koordinates also wanted the ability to co-locate their implementation logic alongside their machines’ definitions, or “blueprints”. This reduces the surface area developers when revisions need to made to their machines. From Taylor: “I love the first class support in XState for side effects. That the side effects can be co-located with my code that triggers it and uses it, instead of having to put them _somewhere else_ like with Redux”.
+
+
+## Handling complex, long-running tasks and edge cases
+
+Being a source control solution for complex datasets, Kart has a naturally complex set of problems to solve as well. Merging, cloning, and loading data all have plenty of considerations, edge cases, and possible failure scenarios, so the team had to take care to address all of them. The ability to map this logic using the Stately Editor proved to be extremely useful.
+
+<EmbedMachine
+  name="Koordinates List View Machine"
+  embedURL="https://stately.ai/registry/editor/1b48bbf1-6f7a-4847-b1e0-47ee5c283371?mode=design&machineId=afa23e5f-11c7-460d-bd98-ff7de694a41d"/>
```
Comment (mellson) [inline blog/2023-11-15-koordinates-and-stately-case-study/index.mdx:59]:
```suggestion
  embedURL="https://stately.ai/registry/editor/embed/1b48bbf1-6f7a-4847-b1e0-47ee5c283371?machineId=afa23e5f-11c7-460d-bd98-ff7de694a41d"/>
```

File: blog/2023-11-15-koordinates-and-stately-case-study/index.mdx (hunk 1 above)
Comment (mellson) [reply blog/2023-11-15-koordinates-and-stately-case-study/index.mdx:44]: I suggest changing this and the next link to be an embed URL.

Comment (gavination) [reply blog/2023-11-15-koordinates-and-stately-case-study/index.mdx:44]: Ah, I thank you for this, @mellson. Didn't know about these parameters.

File: blog/2023-11-15-koordinates-and-stately-case-study/index.mdx [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+date: 2023-11-17
+slug: 2023-11-17-koordinates-and-stately-case-study
+image: /blog/koordinates_logo_rev.png
+---
+
+For more than 15 years, the team at [Koordinates](https://koordinates.com/) has been hard at work reimagining and building streamlined and equitable ways for customers to manage geospatial data. Koordinates have built a massively scaled GIS platform for users to upload their own data, query over 18,000 open datasets, and securely merge openly available data with proprietary information.
+<!--truncate-->
+Unlike competing products that require vendor lock-in and lack compatibility across the entire GIS stack, Koordinates’ offerings are modular: users may use as much or as little of the tooling as they like, and the tooling is interoperable with existing GIS environments.
+
+![Koordinates Logo](./koordinates_logo_rev.png)
```
[the remark this answers is not in the document]
Comment (gavination) [reply blog/2023-11-15-koordinates-and-stately-case-study/index.mdx:15]: oh my! great catch and sorry i didn't see this myself! I've got 2 transparent versions of the image, so i'll lean on the component here.

File: blog/2023-11-15-koordinates-and-stately-case-study/index.mdx [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+
+## Repository State Management
+
+The desktop app Koordinates provides also ships with **[Kart](https://github.com/koordinates/kart),** an open source version control platform built specifically for large geospatial datasets. Being built on top of Git, Kart provides a familiar user experience for developers used to the version control system. Specifically, Kart allows users to:
+
+- **clone** geospatial data to users’ local machines
+- **manage** branches for their datasets - creating, merging, and deleting them
+- **roll back** branches to previous commits, preserving changes made over time
+
+In the case of Kart, many of these operations can be long-running and have **several** edge cases, making Stately’s tooling to visualize, understand, and execute logic from stateful diagrams a massive benefit.
```
[the remark this answers is not in the document]
Comment (gavination) [reply blog/2023-11-15-koordinates-and-stately-case-study/index.mdx:43]: ah there was, as I wanted a bit of an emphasis on how prevalent edge cases are. But, if it wasn't immediately apparent to you, it shouldn't be there.

File: blog/2023-11-15-koordinates-and-stately-case-study/index.mdx [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+
+Handling several logical components does come with some considerations, however. Taylor noted that now that the team had several small components, they needed a way to orchestrate and communicate between them in a cohesive manner. Thankfully, the [XState](https://github.com/statelyai/xstate) framework for composing machines is open source, and the team used it as their base for [`xstate-tree`, an open source framework](https://github.com/koordinates/xstate-tree) for modeling complex UI’s as a tree of XState machines. `xstate-tree` couples React views with state machines and composes them as actors in “slots”, allowing them to be exposed to their respective React views. Learn more about all the features `xstate-tree` provides in their [well-documented GitHub repo](https://github.com/koordinates/xstate-tree).
```
[the remark this answers is not in the document]
Comment (gavination) [reply blog/2023-11-15-koordinates-and-stately-case-study/index.mdx:56]: you are correct. A mistake I make all too often.

File: blog/2023-11-15-koordinates-and-stately-case-study/index.mdx [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+
+While there are other state management platforms for developers, the team at Koordinates also wanted the ability to co-locate their implementation logic alongside their machines’ definitions, or “blueprints”. This reduces the surface area developers when revisions need to made to their machines. From Taylor: “I love the first class support in XState for side effects. That the side effects can be co-located with my code that triggers it and uses it, instead of having to put them _somewhere else_ like with Redux”.
```
[the remark this answers is not in the document]
Comment (gavination) [reply blog/2023-11-15-koordinates-and-stately-case-study/index.mdx:58]: Love this!

File: blog/2023-11-15-koordinates-and-stately-case-study/index.mdx [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+
+
+## Handling complex, long-running tasks and edge cases
+
+Being a source control solution for complex datasets, Kart has a naturally complex set of problems to solve as well. Merging, cloning, and loading data all have plenty of considerations, edge cases, and possible failure scenarios, so the team had to take care to address all of them. The ability to map this logic using the Stately Editor proved to be extremely useful.
+
+<EmbedMachine
```
[the remark this answers is not in the document]
Comment (gavination) [reply blog/2023-11-15-koordinates-and-stately-case-study/index.mdx:65]: thanks so much for grabbing these! Added!

[the remark this answers is not in the document]
Comment (gavination) [reply blog/2023-11-15-koordinates-and-stately-case-study/index.mdx:67]: removed in favor of the embeds! now that scrolling is a bit more reasonable, I think it's the better experience.

File: blog/2023-11-15-koordinates-and-stately-case-study/index.mdx (hunk 3 above)
[the remark this answers is not in the document]
Comment (gavination) [reply blog/2023-11-15-koordinates-and-stately-case-study/index.mdx]: done and done!

File: blog/2023-11-15-koordinates-and-stately-case-study/index.mdx [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+
+## Moving forward with Stately
+
+By building with Stately’s tooling, the Koordinates team not only simplified their UI and repository state management, but they in turn gave back to the technical community with `xstate-tree`, allowing React developers to better wrangle hosts of machines. Koordinates and Stately are aligned on many core goals: enhancing productivity for their users, strongly committing to open source, and facilitating team collaboration. So, the Stately team was eager for feedback and have committed to the development of features that enhance collaboration without compromise, like GitHub synchronization to respect the repo as the single source of truth as well as system-wide statechart visualization to facilitate deeper interdisciplinary communication and understanding.
```
[the remark this answers is not in the document]
Comment (gavination) [reply blog/2023-11-15-koordinates-and-stately-case-study/index.mdx]: I see. A nice call to action. Keeping this issue open to get it "just so", and would love feedback on this second swing.

File: blog/2023-11-15-koordinates-and-stately-case-study/index.mdx [hunk 9]
```diff
@@ -0,0 +1,67 @@
+---
+title: 'Koordinates x Stately: geospatial data management case study'
+description: Making geographic information systems accessible and collaborative. A Stately and Koordinates case study.
+authors: [gavin]
+tags: [case study, study, koordinates, stately, state machines, machines, gis, geospatial]
+date: 2023-11-17
+slug: 2023-11-17-koordinates-and-stately-case-study
+image: /blog/koordinates_logo_rev.png
+---
+
```
[the remark this answers is not in the document]
Comment (gavination) [reply blog/2023-11-15-koordinates-and-stately-case-study/index.mdx:10]: I like a nice one-liner. They're always tricky though! How is this guy? 

```When it came to navigating the complexities of business context and application state, Koordinates found their solution in Stately's tools, reshaping their coding practices and ensuring a seamless user experience."
```

[the remark this answers is not in the document]
Comment (laurakalbag) [reply blog/2023-11-15-koordinates-and-stately-case-study/index.mdx:10]: Love it!

File: blog/2023-11-15-koordinates-and-stately-case-study/index.mdx [hunk 10]
```diff
@@ -0,0 +1,81 @@
...
+## Moving forward with Stately
+
+By building with Stately’s tooling, the Koordinates team not only simplified their UI and repository state management, but they in turn gave back to the technical community with `xstate-tree`, allowing React developers to better wrangle hosts of machines. Koordinates and Stately are aligned on many core goals: enhancing productivity for their users, strongly committing to open source, and facilitating team collaboration. Naturally, the Stately team was eager for feedback and have committed to the development of features that enhance collaboration without compromise. This includes GitHub synchronization to maintain the repo as the authoritative source and system-wide statechart visualization to facilitate deeper interdisciplinary communication and understanding. Koordinates saw a significant leap in efficiency and team collaboration with Stately's tools.[ Book a demo with us today](https://calendly.com/d/yc8-3hq-rpc/request-a-demo) to learn how Stately can empower your team to build better software, faster.
```
Comment (laurakalbag) [inline blog/2023-11-15-koordinates-and-stately-case-study/index.mdx:81]:
```suggestion
By building with Stately’s tooling, the Koordinates team not only simplified their UI and repository state management, but they in turn gave back to the technical community with `xstate-tree`, allowing React developers to better wrangle hosts of machines. Koordinates and Stately are aligned on many core goals: enhancing productivity for their users, strongly committing to open source, and facilitating team collaboration. Naturally, the Stately team was eager for feedback and have committed to the development of features that enhance collaboration without compromise. This includes GitHub synchronization to maintain the repo as the authoritative source and system-wide statechart visualization to facilitate deeper interdisciplinary communication and understanding. 

Koordinates saw a significant leap in efficiency and team collaboration with Stately's tools. [Book a demo with us today](https://calendly.com/d/yc8-3hq-rpc/request-a-demo) to learn how Stately can empower your team to build better software, faster.
```

Comment (laurakalbag) [reply blog/2023-11-15-koordinates-and-stately-case-study/index.mdx:81]: Perfect, just broken it into a couple of bitesize chunks.

Comment (gavination) [reply blog/2023-11-15-koordinates-and-stately-case-study/index.mdx:81]: Love it!

**Review by laurakalbag — APPROVED** [review]

**Pull request merged**